src/contributors/ComputeContributor.ts

```diff
@@ -54,28 +51,29 @@ export class ComputeContributor extends Contributor {
         return jsonSchema;
     }
 
-    public fetchData(collaborationEvent: CollaborationEvent): Observable<ComputationResponse> {
-        const computationResponse = this.collaborativeSearcheService.resolveButNotComputation([projType.compute,
-        <ComputationRequest>{ field: this.field, metric: ComputationRequest.MetricEnum[this.metric.toUpperCase()] }],
-            this.collaborativeSearcheService.collaborations, this.identifier, {}, false, this.cacheDuration);
+    public fetchData(collaborationEvent: CollaborationEvent): Observable<Array<ComputationResponse>> {
+
+        const computationResponse: Observable<Array<ComputationResponse>> = forkJoin(this.metrics.map(m => {
+            return this.collaborativeSearcheService.resolveButNotComputation([projType.compute,
+            <ComputationRequest>{ field: m.field, metric: ComputationRequest.MetricEnum[m.metric.toUpperCase()] }],
+                this.collaborativeSearcheService.collaborations, this.identifier, {}, false, this.cacheDuration);
+        }));
 
         if (collaborationEvent.id !== this.identifier || collaborationEvent.operation === OperationEnum.remove) {
             return computationResponse;
         } else {
             return from([]);
         }
+
     }
 
-    public computeData(data: ComputationResponse): ComputationResponse {
+    public computeData(data: Array<ComputationResponse>): Array<ComputationResponse> {
         return data;
     }
 
-    public setData(data: ComputationResponse): any {
-        const result = data.value;
-        let resultValue = result;
-        if (this.process && this.process.trim().length > 0) {
-            resultValue = eval(this.process);
-        }
+    public setData(data: Array<ComputationResponse>): any {
+        const m = data.map(d => d.value);
```

Comment:
`m` is never used

Reply:
I invite you to read the json schema, m is used in the function inside the eva

Reply:
Ok, I see it